examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+    "        return <button> {my_message} </button> \n",
+    "        }};\"\"\"\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "944363e3-92b1-4ee6-8e16-1a2f6879fefb",
+   "metadata": {},
+   "source": [
+    "Instead, we can **use traitlets.**  \n",
+    "traitlets are variables that can be used both in JavaScript and python.  \n",
+    "Note that it has to be initilized at `MyButton({my_message})`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e27dc18-82f2-4598-9117-9e0cfec49f99",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "from traitlets import Unicode\n",
+    "\n",
+    "# ⭐⭐⭐  This is good practice again ⭐⭐⭐\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    my_message = Unicode(\"Hi There\").tag(sync=True)\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton({ my_message }) {\n",
+    "    return <button> {my_message} </button>;\n",
+    "    };\"\"\"\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e2fb6dcb-4477-4ff5-a7d0-5e0f209d83cf",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# we can use this traitlets also as parameters\n",
+    "MyExampleWidget(my_message = \"Super Button\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12f3691c-6d99-40ac-b80d-1db9b6f5d78c",
+   "metadata": {},
+   "source": [
+    "Now comes an example of a button that changes the postion using traitlets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "56c9e4d7-03fc-4a30-81f8-83d63bab170a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "from traitlets import Int\n",
+    "\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    my_width = Int(23).tag(sync=True)\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton({ my_width }) {\n",
+    "      return (\n",
+    "        <button\n",
+    "          style={{\n",
+    "            position: \"relative\",\n",
+    "            width: my_width,\n",
+    "            height: 30,\n",
+    "          }}\n",
+    "        >\n",
+    "          {\" \"}\n",
+    "          Width of {my_width} px{\" \"}\n",
+    "        </button>\n",
+    "      );\n",
+    "    }\"\"\"\n",
+    "MyExampleWidget(my_width=300)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4dba603f-8052-4d8d-845d-d28029e17c35",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MyExampleWidget(my_width=600)\n",
+    "#MyExampleWidget(my_width=\"Hello\") # <- this will throw an error, as my_width expects an integer"
```

Comment:
This will throw a python error as my_width expects an integer.

throwing apython error is a feature.  It's easier to understand than the corresponding JS error or worse non functioning CSS that would occur otherwise.   This might be worth mentioning for users.